data/core/doc/init.lua

```diff
@@ -509,6 +511,14 @@ function Doc:text_input(text, idx)
     if line1 ~= line2 or col1 ~= col2 then
       self:delete_to_cursor(sidx)
     end
+
+    if self.overwrite
+    and col1 ~= #self.lines[line1]
+    and #text:gsub('\n', '') ~= 0 then
+      local _, next_col = translate.next_char(self, line1, col1)
+      self:remove(line1, col1, line1, next_col)
+    end
+
```

Comment:
`text` can be more than 1 character, and we should delete the same number of characters (not bytes) we're writing.